sekoia_automation/http/aio/http_client.py

```diff
@@ -0,0 +1,210 @@
+                yield self._session
+        else:
+            yield self._session
+
+    @asynccontextmanager
+    async def get_retry(
+        self, url: str, *args: Any, **kwargs: Optional[Any]
+    ) -> AsyncGenerator[ClientResponse, None]:
+        """
+        Get callable.
+
+        Args:
+            url: str
+            args: Any
+            kwargs: Optional[Any]
+
+        Returns:
+            ClientResponse:
+        """
+        async with self.request_retry("GET", url, *args, **kwargs) as result:
+            yield result
+
+    @asynccontextmanager
+    async def post_retry(
+        self, url: str, *args: Any, **kwargs: Optional[Any]
+    ) -> AsyncGenerator[ClientResponse, None]:
+        """
+        Post callable.
+
+        Args:
+            url: str
+            args: Any
+            kwargs: Optional[Any]
+
+        Returns:
+            ClientResponse:
+        """
+        async with self.request_retry("POST", url, *args, **kwargs) as result:
+            yield result
+
+    @asynccontextmanager
+    async def put_retry(
+        self, url: str, *args: Any, **kwargs: Optional[Any]
+    ) -> AsyncGenerator[ClientResponse, None]:
+        """
+        Put callable.
+
+        Args:
+            url: str
+            args: Any
+            kwargs: Optional[Any]
+
+        Returns:
+            ClientResponse:
+        """
+        async with self.request_retry("PUT", url, *args, **kwargs) as response:
+            yield response
+
+    @asynccontextmanager
+    async def delete_retry(
+        self, url: str, *args: Any, **kwargs: Optional[Any]
+    ) -> AsyncGenerator[ClientResponse, None]:
+        """
+        Delete callable.
+
+        Args:
+            url: str
+            args: Any
+            kwargs: Optional[Any]
+
+        Returns:
+            ClientResponse:
+        """
+        async with self.request_retry("DELETE", url, *args, **kwargs) as response:
+            yield response
+
+    @asynccontextmanager
+    async def patch_retry(
+        self, url: str, *args: Any, **kwargs: Optional[Any]
+    ) -> AsyncGenerator[ClientResponse, None]:
+        """
+        Patch callable.
+
+        Args:
+            url: str
+            args: Any
+            kwargs: Optional[Any]
+
+        Returns:
+            ClientResponse:
+        """
+        async with self.request_retry("PATCH", url, *args, **kwargs) as response:
+            yield response
+
+    @asynccontextmanager
+    async def head_retry(
+        self, url: str, *args: Any, **kwargs: Optional[Any]
+    ) -> AsyncGenerator[ClientResponse, None]:
+        """
+        Head callable.
+
+        Args:
+            url: str
+            args: Any
+            kwargs: Optional[Any]
+
+        Returns:
+            ClientResponse:
+        """
+        async with self.request_retry("HEAD", url, *args, **kwargs) as response:
+            yield response
+
+    @asynccontextmanager
+    async def request_retry(
```

Comment:
Same: why the suffix `_retry`?

Reply:
Retry means that this method perform a request with retry. 
The same for all methods like:
get_retry, post_retry etc.
Why I did in such way is because if `retry` is not properly configured, we should add additional if else to base method. However I can change it